File: android/app/src/main/kotlin/com/example/sound_detective/channel/ChannelNames.kt
package com.example.sound_detective.channel

/** Mirrors lib/core/constants/channel_names.dart — keep in sync. */
object ChannelNames {
    const val CONTROL = "sounddetective/control"
    const val EVENTS = "sounddetective/events"
    const val SERVICE_STATUS = "sounddetective/service_status"
}

object ControlMethod {
    const val START_FOREGROUND_SERVICE = "startForegroundService"
    const val STOP_FOREGROUND_SERVICE = "stopForegroundService"
    const val IS_SERVICE_RUNNING = "isServiceRunning"

    const val CHECK_NOTIFICATION_ACCESS_GRANTED = "checkNotificationAccessGranted"
    const val OPEN_NOTIFICATION_ACCESS_SETTINGS = "openNotificationAccessSettings"

    const val CHECK_USAGE_ACCESS_GRANTED = "checkUsageAccessGranted"
    const val OPEN_USAGE_ACCESS_SETTINGS = "openUsageAccessSettings"

    const val CHECK_BLUETOOTH_PERMISSION_GRANTED = "checkBluetoothPermissionGranted"
    const val REQUEST_BLUETOOTH_PERMISSION = "requestBluetoothPermission"

    const val CHECK_BATTERY_OPTIMIZATION_EXEMPT = "checkBatteryOptimizationExempt"
    const val REQUEST_BATTERY_OPTIMIZATION_EXEMPTION = "requestBatteryOptimizationExemption"

    const val GET_RECENT_EVENTS = "getRecentEvents"
    const val RECONSTRUCT_FOREGROUND_APP = "reconstructForegroundApp"
    const val CONSUME_LAUNCH_ACTION = "consumeLaunchAction"
    const val GET_CURRENT_RINGER_MODE = "getCurrentRingerMode"
}

File: lib/core/constants/channel_names.dart
/// Names for the platform channels shared between Dart and the native
/// Android side. Keep in sync with the Kotlin constants in
/// `android/app/src/main/kotlin/.../channel/ChannelNames.kt`.
class ChannelNames {
  ChannelNames._();

  static const String control = 'sounddetective/control';
  static const String events = 'sounddetective/events';
  static const String serviceStatus = 'sounddetective/service_status';
}

/// Method names used on [ChannelNames.control].
class ControlMethods {
  ControlMethods._();

  static const String startForegroundService = 'startForegroundService';
  static const String stopForegroundService = 'stopForegroundService';
  static const String isServiceRunning = 'isServiceRunning';

  static const String checkNotificationAccessGranted =
      'checkNotificationAccessGranted';
  static const String openNotificationAccessSettings =
      'openNotificationAccessSettings';

  static const String checkUsageAccessGranted = 'checkUsageAccessGranted';
  static const String openUsageAccessSettings = 'openUsageAccessSettings';

  static const String checkBluetoothPermissionGranted =
      'checkBluetoothPermissionGranted';
  static const String requestBluetoothPermission =
      'requestBluetoothPermission';

  static const String checkBatteryOptimizationExempt =
      'checkBatteryOptimizationExempt';
  static const String requestBatteryOptimizationExemption =
      'requestBatteryOptimizationExemption';

  static const String getRecentEvents = 'getRecentEvents';
  static const String reconstructForegroundApp = 'reconstructForegroundApp';
  static const String consumeLaunchAction = 'consumeLaunchAction';
  static const String getCurrentRingerMode = 'getCurrentRingerMode';
}
